internal/ansible/ansible_manager.go

```diff
@@ -0,0 +1,222 @@
+	return &AnsibleManager{
+		dispatcherClient:                 dispatcherClient,
+		ansiblePlaybookConnectionOptions: ansiblePlaybookConnectionOptions,
+		ansiblePlaybookOptions:           ansiblePlaybookOptions,
+		playbook:                         &playbook.AnsiblePlaybookCmd{},
+	}
+}
+
+func MissingAttributeError(attribute string, metadata map[string]string) error {
+	return fmt.Errorf(missingAttributeMsg(attribute, metadata))
+}
+
+func missingAttributeMsg(attribute string, metadata map[string]string) string {
+	return fmt.Sprintf("missing attribute %s in message metadata %v", attribute, metadata)
+}
+
+func invalidAttributeMsg(attribute string, value string) string {
+	return fmt.Sprintf("invalid attribute %s = %s in message", attribute, value)
+}
+
+func (a *AnsibleManager) HandlePlaybook(d *pb.Data) error {
+	// required fields
+	reqFields := &RequiredFields{}
+	payloadStr := string(d.Content)
+
+	found := false
+	dataDir := "/tmp"
+
+	var err error
+	responseTo := d.MessageId
+	if payloadStr == "" {
+		return fmt.Errorf("missing playbook string in message")
+	}
+	if reqFields.crcDispatcherCorrelationId, found = d.GetMetadata()[crcDispatcherAttribute]; !found {
+		return MissingAttributeError(crcDispatcherAttribute, d.GetMetadata())
+	}
+	if response_interval, found := d.GetMetadata()[responseIntervalAttribute]; !found {
+		log.Warnf(missingAttributeMsg(responseIntervalAttribute, d.GetMetadata())+" Defaulting to "+strconv.Itoa(defaultResponseInterval), d.GetMetadata())
+		reqFields.responseInterval = defaultResponseInterval
+	} else {
+		reqFields.responseInterval, err = strconv.ParseUint(response_interval, 10, 64)
+		if err != nil {
+			log.Warnf(invalidAttributeMsg(responseIntervalAttribute, d.GetMetadata()[responseIntervalAttribute]+" Defaulting to "+strconv.Itoa(defaultResponseInterval)))
+			reqFields.responseInterval = defaultResponseInterval
+		}
+	}
+	if reqFields.returnUrl, found = d.GetMetadata()[returnUrlAttribute]; !found {
+		return fmt.Errorf(missingAttributeMsg(returnUrlAttribute, d.GetMetadata()))
+	}
+
+	playbookYamlFile := path.Join(dataDir, "ansible_playbook_"+d.MessageId+".yml")
+	err = os.WriteFile(playbookYamlFile, d.GetContent(), 0777)
+
+	if err != nil {
+		return fmt.Errorf("cannot create ansible playbook yaml file %s. Error: %v", playbookYamlFile, err)
+	}
+
+	/*
+	*******************************************
+	*  TODO : Verify playbook
+	*******************************************
+	 */
+
+	eventList := dispatcher.NewEventList()
+	// required event for cloud connector
+	onStart := dispatcher.ExecutorOnStart(reqFields.crcDispatcherCorrelationId, "")
+	eventList.AddEvent(onStart)
+
+	playbook := &playbook.AnsiblePlaybookCmd{
+		Playbooks:         []string{playbookYamlFile},
+		ConnectionOptions: ansiblePlaybookConnectionOptions,
+		Options:           ansiblePlaybookOptions,
+		// Exec:              events.AddEvent, ??
```

Comment:
Is there a way to invoke `AddEvent` function every time an Ansible event is received? (Similary to [rhc-worker-playbook](https://github.com/RedHatInsights/rhc-worker-playbook/blob/acd51a1cab05322e38d9ce68113af61ddcc7de54/rhc_worker_playbook/server.py#L212) )

Reply:
We could use an interface similar to this [class](https://github.com/RedHatInsights/rhc-worker-playbook/blob/acd51a1cab05322e38d9ce68113af61ddcc7de54/rhc_worker_playbook/server.py#L102) and later one send it [back](https://github.com/RedHatInsights/rhc-worker-playbook/blob/acd51a1cab05322e38d9ce68113af61ddcc7de54/rhc_worker_playbook/server.py#L66) to the operator. The question is how to expose it to the users.

Reply:
Well there is no way using the ansible-playbook command AFAIK, only using ansible-runner. The only way is to write custom [callback plugin](https://docs.ansible.com/ansible/latest/plugins/callback.html). Or integrate ansible-runner command inside the go-ansible.

Reply:
Thanks @machacekondra for your suggestion! 
I created a custom executor to handle the runner results. Next step is to customize di StdOutCallback. Let's see if I'll manage to make it work.